exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -375,3 +381,21 @@ def _create_info_metric(
         info = InfoMetricFamily(name, description, labels=attributes)
         info.add_metric(labels=list(attributes.keys()), value=attributes)
         return info
+
+
+class _AutoPrometheusMetricReader(PrometheusMetricReader):
+    """Thin wrapper around PrometheusMetricReader used for opentelemetry_metrics_exporter
+
+    This allows users to use the prometheus exporter with opentelemetry-instrument. It handles
+    starting the Prometheus http server on the the correct port and host.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Default values are specified in
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/v1.24.0/specification/configuration/sdk-environment-variables.md#prometheus-exporter
+        start_http_server(
+            port=int(environ.get(OTEL_EXPORTER_PROMETHEUS_PORT, "9464")),
```

Comment:
Just to clarify, the if exporter is used with auto-instrumentation, a server will be started for them, whereas if user were to just use the exporter manually, it assumes that a server is already running?

Reply:
Yes

> it assumes that a server is already running?

prometheus client lib actually allows them to start it after this code. They can do any of these options https://github.com/prometheus/client_python#exporting